modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java

```diff
@@ -0,0 +1,359 @@
+
+import io.swagger.codegen.*;
+import io.swagger.models.*;
+import io.swagger.models.parameters.*;
+import io.swagger.models.properties.*;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.util.*;
+import java.io.File;
+
+public class GatlingCodegen extends AbstractScalaCodegen implements CodegenConfig {
+
+    // source folder where to write the files
+    protected String sourceFolder = "src" + File.separator + "gatling" + File.separator + "scala";
+    protected String resourceFolder = "src" + File.separator + "gatling" + File.separator + "resources";
+    protected String confFolder = resourceFolder + File.separator + "conf";
+    protected String dataFolder = resourceFolder + File.separator + "data";
+    protected String apiVersion = "1.0.0";
+
+    /**
+     * Configures the type of generator.
+     *
+     * @return the CodegenType for this generator
+     * @see io.swagger.codegen.CodegenType
+     */
+    public CodegenType getTag() {
+        return CodegenType.CLIENT;
+    }
+
+    /**
+     * Configures a friendly name for the generator.  This will be used by the generator
+     * to select the library with the -l flag.
+     *
+     * @return the friendly name for the generator
+     */
+    public String getName() {
+        return "gatling";
+    }
+
+    /**
+     * Returns human-friendly help for the generator.  Provide the consumer with help
+     * tips, parameters here
+     *
+     * @return A string value for the help message
+     */
+    public String getHelp() {
+        return "Generates a gatling simulation library.";
+    }
+
+    public GatlingCodegen() {
+        super();
+
+        // set the output folder here
+        outputFolder = "generated-code/gatling";
+
+        /**
+         * Api classes.  You can write classes for each Api file with the apiTemplateFiles map.
+         * as with models, add multiple entries with different extensions for multiple files per
+         * class
+         */
+        apiTemplateFiles.put(
+                "api.mustache",   // the template to use
+                "Simulation.scala");       // the extension for each file to write
+
+        modelTemplateFiles.put("model.mustache", ".scala");
+
+        /**
+         * Template Location.  This is the location which templates will be read from.  The generator
+         * will use the resource stream to attempt to read the templates.
+         */
+        templateDir = "Gatling";
+
+        /**
+         * Api Package.  Optional, if needed, this can be used in templates
+         */
+        apiPackage = "io.swagger.client.api";
+
+        /**
+         * Model Package.  Optional, if needed, this can be used in templates
+         */
+        modelPackage = "io.swagger.client.model";
+
+        /**
+         * Reserved words.  Override this with reserved words specific to your language
+         */
+        reservedWords = new HashSet<String>(
```

Comment:
I don't think this is needed.

Reply:
you may be right, will test with out, i think the AbstractScalaCodegen takes care of this for me now.